src/platforms/stlink/platform.h

```diff
@@ -78,6 +83,19 @@ extern bool debug_bmp;
 #define SWD_CR_MULT (1U << ((14U - 8U) << 2U))
 
 #define TMS_SET_MODE() gpio_set_mode(TMS_PORT, GPIO_MODE_OUTPUT_2_MHZ, GPIO_CNF_OUTPUT_PUSHPULL, TMS_PIN);
+
+#ifdef STLINK_V2_ISOL
+/* The ISOL variant floats SWDIO with GPIO A1 */
+#define SWDIO_MODE_FLOAT()      \
+	do {                        \
+		gpio_set(GPIOA, GPIO1); \
```

Comment:
Ideally this and the clear below would use the unpacked/inlined version of the code this calls, but this works. Just to clarify though - PA1 is the SWDIO direction setting pin? Want to be sure we've understood the purpose of this code.

Reply:
Good point - I can inline the code

Reply:
* Interesting when you put it that way. Does PA1 high *switch* the direction to input of levelshifter between T_SWDIO_OUT and PB14 coupler channel output, or *disconnect* it? I wonder if the shifter as input and coupler as output will fight (as the target drives its data against a fixed-level BMF TMS). In that case, consider also switching SWDIO_OUT to digital input mode. SWDIO_IN can and should stay.
* PA1 might be `TMS_DIR_PIN` (see the other comment). What about `TMS_SET_MODE()`? jtagtap will call it, is it identical in function to `SWDIO_MODE_DRIVE()`? I expect that after a SWD detach, your new platform may fail to drive JTAG TMS because of stuck direction pin. Please post logs with JTAG scan tests.

Reply:
About the SWDIO multiplexing - the probe uses Si8420BB RF isolators which are fixed direction. If we ignore the isolators the schematic looks like this: 
![image](https://github.com/blackmagic-debug/blackmagic/assets/34950964/0b046053-e657-434a-8ace-8e7b280b0708)
PB14 and PA1 are always forced to output and PB12 is always forced to input because of the unidirectional RF isolators used to interface between the 2G125 IC and the STM32F01

SWDCLK should be similar, but I think PA4 is used to enable the output propagation or set it to high impedance

Reply:
> About the SWDIO multiplexing - the probe uses Si8420BB RF isolators which are fixed direction. If we ignore the isolators the schematic looks like this: ![image](https://private-user-images.githubusercontent.com/34950964/296870492-0b046053-e657-434a-8ace-8e7b280b0708.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU3NTk2MjgsIm5iZiI6MTcwNTc1OTMyOCwicGF0aCI6Ii8zNDk1MDk2NC8yOTY4NzA0OTItMGIwNDYwNTMtZTY1Ny00MzRhLThhY2UtOGU3YjI4MGIwNzA4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIwVDE0MDIwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYyYTNlODExMzZmYWJhOTMyNDA3NTNhMjI4MTlkYWM5ZTRjMGQ1ODAzZGM4NmI3Nzc5MGM4YmI3MWNjMjA0N2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9PBpkEhC4_b1g7nkQQjHHWqgt21pXeJmQWVpq-bcZxo) PB14 and PA1 are always forced to output and PB12 is always forced to input because of the unidirectional RF isolators used to interface between the 2G125 IC and the STM32F01
> 
> SWDCLK should be similar, but I think PA4 is used to enable the output propagation or set it to high impedance

This was hidden after the changes, but may be a good idea to be easily accessible